pennylane/qinfo/transforms.py

```diff
@@ -396,3 +396,163 @@ def wrapper(*args, **kwargs):
+
+        @qml.qnode(dev)
+        def circuit_rx(x, y):
+            qml.RX(x, wires=0)
+            qml.RZ(y, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit_ry(y):
+            qml.RY(y, wires=0)
+            return qml.state()
+
+    >>> qml.qinfo.fidelity(circuit_rx, circuit_ry, wires0=[0], wires1=[0])((0.1, 0.3), (0.2))
+    0.9905158135644924
+
+    It is also possible to use QNodes that do not depend on any parameters. When it is the case for the first QNode, you
+    need to pass an empty tuple as an argument for the first QNode.
+
+    .. code-block:: python
+
+        dev = qml.device('default.qubit', wires=1)
+
+        @qml.qnode(dev)
+        def circuit_rx():
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit_ry(x):
+            qml.RY(x, wires=0)
+            return qml.state()
+
+    >>> qml.qinfo.fidelity(circuit_rx, circuit_ry, wires0=[0], wires1=[0])((), (0.2))
+    0.9900332889206207
+
+    On the other hand, if the second QNode is the one that does not depend on parameters then a single tuple can also be
+    passed:
+
+    >>> qml.qinfo.fidelity(circuit_ry, circuit_rx, wires0=[0], wires1=[0])((0.2))
+    0.9900332889206207
+
+    The `qml.qinfo.fidelity` transform is also differentiable and you can use the gradient in the different frameworks
+    with backpropagation, the following example uses `jax` and `backprop`.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit0(x):
+            qml.RX(x, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev, interface="jax")
+        def circuit1():
+            qml.PauliZ(wires=0)
+            return qml.state()
+
+    >>> jax.grad(qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0]))((jax.numpy.array(0.3)))
+    -0.14776011
+
+    """
+
+    if len(wires0) != len(wires1):
+        raise qml.QuantumFunctionError("The two states must have the same number of wires.")
+
+    # Get the state vector if all wires are selected
+    if len(wires0) == len(qnode0.device.wires):
+        state_qnode0 = qnode0
+    else:
+        state_qnode0 = qml.qinfo.reduced_dm(qnode0, wires=wires0)
+
+    # Get the state vector if all wires are selected
+    if len(wires1) == len(qnode1.device.wires):
+        state_qnode1 = qnode1
+    else:
+        state_qnode1 = qml.qinfo.reduced_dm(qnode1, wires=wires1)
+
+    def evaluate_fidelity(all_args0=None, all_args1=None):
+        """Wrapper used for evaluation of the fidelity between two states computed from QNodes. It allows giving
+        the args and kwargs to each :class:`.QNode`.
+
+        Args:
+            all_args0 (tuple): Tuple containing the arguments (*args, **kwargs) of the first :class:`.QNode`.
+            all_args1 (tuple): Tuple containing the arguments (*args, **kwargs) of the second :class:`.QNode`.
```

Comment:
It doesn't look like `kwargs` is supported in the code below (since we're unpacking the entire `all_args0` as a tuple).